File: JsonSchemaAndStructs/KiteConnect-Kotlin-Structs/ohlc.kt
// To parse the JSON, install Klaxon and do:
//
//   val ohlc = Ohlc.fromJson(jsonString)

package quicktype

import com.beust.klaxon.*

private val klaxon = Klaxon()

data class Ohlc (
    @Json(name = "\$ref")
    val ref: String,

    @Json(name = "\$schema")
    val schema: String,

    val definitions: Definitions
) {
    public fun toJson() = klaxon.toJsonString(this)

    companion object {
        public fun fromJson(json: String) = klaxon.parse<Ohlc>(json)
    }
}

data class Definitions (
    @Json(name = "Data")
    val data: Data,

    @Json(name = "NseInfy")
    val nseInfy: NseInfyClass,

    @Json(name = "Ohlc")
    val ohlc: OhlcClass,

    @Json(name = "OhlcClass")
    val ohlcClass: OhlcClassClass
)

data class Data (
    val additionalProperties: Boolean,
    val properties: DataProperties,
    val required: List<String>,
    val title: String,
    val type: String
)

data class DataProperties (
    @Json(name = "NSE:INFY")
    val nseInfy: NseInfy
)

data class NseInfy (
    @Json(name = "\$ref")
    val ref: String
)

data class NseInfyClass (
    val additionalProperties: Boolean,
    val properties: NseInfyProperties,
    val required: List<String>,
    val title: String,
    val type: String
)

data class NseInfyProperties (
    @Json(name = "instrument_token")
    val instrumentToken: InstrumentToken,

    @Json(name = "last_price")
    val lastPrice: InstrumentToken,

    val ohlc: NseInfy
)

data class InstrumentToken (
    val type: String
)

data class OhlcClass (
    val additionalProperties: Boolean,
    val properties: OhlcProperties,
    val required: List<String>,
    val title: String,
    val type: String
)

data class OhlcProperties (
    val data: NseInfy,
    val status: InstrumentToken
)

data class OhlcClassClass (
    val additionalProperties: Boolean,
    val properties: OhlcClassProperties,
    val required: List<String>,
    val title: String,
    val type: String
)

data class OhlcClassProperties (
    val close: InstrumentToken,
    val high: InstrumentToken,
    val low: InstrumentToken,
    val open: InstrumentToken
)

File: JsonSchemaAndStructs/KiteConnect-Rust-Structs/ohlc.rs
// Example code that deserializes and serializes the model.
// extern crate serde;
// #[macro_use]
// extern crate serde_derive;
// extern crate serde_json;
//
// use generated_module::[object Object];
//
// fn main() {
//     let json = r#"{"answer": 42}"#;
//     let model: [object Object] = serde_json::from_str(&json).unwrap();
// }

extern crate serde_derive;

#[derive(Serialize, Deserialize)]
pub struct Ohlc {
    #[serde(rename = "$ref")]
    ohlc_ref: String,

    #[serde(rename = "$schema")]
    schema: String,

    #[serde(rename = "definitions")]
    definitions: Definitions,
}

#[derive(Serialize, Deserialize)]
pub struct Definitions {
    #[serde(rename = "Data")]
    data: Data,

    #[serde(rename = "NseInfy")]
    nse_infy: NseInfyClass,

    #[serde(rename = "Ohlc")]
    ohlc: OhlcClass,

    #[serde(rename = "OhlcClass")]
    ohlc_class: OhlcClassClass,
}

#[derive(Serialize, Deserialize)]
pub struct Data {
    #[serde(rename = "additionalProperties")]
    additional_properties: bool,

    #[serde(rename = "properties")]
    properties: DataProperties,

    #[serde(rename = "required")]
    required: Vec<String>,

    #[serde(rename = "title")]
    title: String,

    #[serde(rename = "type")]
    data_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct DataProperties {
    #[serde(rename = "NSE:INFY")]
    nse_infy: NseInfy,
}

#[derive(Serialize, Deserialize)]
pub struct NseInfy {
    #[serde(rename = "$ref")]
    nse_infy_ref: String,
}

#[derive(Serialize, Deserialize)]
pub struct NseInfyClass {
    #[serde(rename = "additionalProperties")]
    additional_properties: bool,

    #[serde(rename = "properties")]
    properties: NseInfyProperties,

    #[serde(rename = "required")]
    required: Vec<String>,

    #[serde(rename = "title")]
    title: String,

    #[serde(rename = "type")]
    nse_infy_class_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct NseInfyProperties {
    #[serde(rename = "instrument_token")]
    instrument_token: InstrumentToken,

    #[serde(rename = "last_price")]
    last_price: InstrumentToken,

    #[serde(rename = "ohlc")]
    ohlc: NseInfy,
}

#[derive(Serialize, Deserialize)]
pub struct InstrumentToken {
    #[serde(rename = "type")]
    instrument_token_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct OhlcClass {
    #[serde(rename = "additionalProperties")]
    additional_properties: bool,

    #[serde(rename = "properties")]
    properties: OhlcProperties,

    #[serde(rename = "required")]
    required: Vec<String>,

    #[serde(rename = "title")]
    title: String,

    #[serde(rename = "type")]
    ohlc_class_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct OhlcProperties {
    #[serde(rename = "data")]
    data: NseInfy,

    #[serde(rename = "status")]
    status: InstrumentToken,
}

#[derive(Serialize, Deserialize)]
pub struct OhlcClassClass {
    #[serde(rename = "additionalProperties")]
    additional_properties: bool,

    #[serde(rename = "properties")]
    properties: OhlcClassProperties,

    #[serde(rename = "required")]
    required: Vec<String>,

    #[serde(rename = "title")]
    title: String,

    #[serde(rename = "type")]
    ohlc_class_class_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct OhlcClassProperties {
    #[serde(rename = "close")]
    close: InstrumentToken,

    #[serde(rename = "high")]
    high: InstrumentToken,

    #[serde(rename = "low")]
    low: InstrumentToken,

    #[serde(rename = "open")]
    open: InstrumentToken,
}
